native/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxTreeGenerator.java

```diff
@@ -193,10 +206,32 @@ public static ArrayList<Entity> getEntityRecord(Path filePath, Optional<String>
                 }
             }
         }
-        return entityArray;
+        Map<String, ArrayList> map = new HashMap();
+        map.put("entityArray", entityArray);
+        map.put("entityMembers", entityMembers);
```

Comment:
Rather than a Map, it is better to have a class `EntityMetadata` perhaps? IMO its much more clear for readability.

Reply:
updated in [a7041fd](https://github.com/ballerina-platform/persist-tools/pull/31/commits/a7041fdbccf9572781cd185694e8cb15352d4e0b)